packages/client/src/client.js

```diff
@@ -504,6 +507,25 @@ export class PercyClient {
     await this.finalizeComparison(comparison.data.id);
     return comparison;
   }
+
+  // decides project type
+  tokenType() {
+    let token = this.token || this.env.token || '';
```

Comment:
```suggestion
    let token = this.token || this.env.token || '';
```

lets call `getToken` function, with an argument `{ raiseIfMissing: false}` ( default would be true)
so we still use the function to fetch token.

Reply:
`getToken()` is already a defined function. If you meant passing in `tokenType` a variable even then it won't solve much issue as we would end up writing if/else logic only.

Reply:
what I meant is

`getToken({{ raiseIfMissing: true}})` this would be by default definition of the function, so it won't affect existing users
and wrap the `raise error` with this condition.

in `tokenType()` function you'll call `getToken({raiseIfMissing: false})` so even if token is missing it won't raise an error.

Reply:
ack, lets update spec for `getToken`